apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx

```diff
@@ -419,6 +432,25 @@ const PoolCandidatesTable = ({
         },
       },
     ),
+    ...(currentPool
+      ? []
+      : [
+          columnHelper.accessor(
+            ({ poolCandidate: { pool } }) => getFullPoolTitleLabel(intl, pool),
+            {
+              id: "process",
+              header: intl.formatMessage(processMessages.process),
+              sortingFn: normalizedText,
```

Comment:
Pool candidates table is API driven, so the search and sort has the external factor. May need to be added into this function so sorting by process works. May also want to merge main first to incorporate my recent changes to this sort

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/bd39dc10-04ef-41d9-b78f-792cfcdec769)

Reply:
This is proving to be incredibly tricky since it is a property of a `jsonb` column on a related table. I'm going to keep playing around to see if I can get it but we may need to spin sorting off into a separate ticket 😬

Reply:
Ah shoot, I forgot that it would be a bit trickier than just that. Thought it was just a string.
I think for now just disable the sort and leave it at that?

Reply:
Okay, after some help from @tristan-orourke I settled on just creating a new scope for this query to work around the issues with ordering by related jsonb properties. This was done in two commits. I tried adding some documentation so we know why we did this weird thing 😅 Let me know if it does not make sense!

 - [add custom orderby scope for json relationship aggregates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9320/commits/4e244574dcde4994f2701ee53515154a51d8b6d5)
 - [implement custom order by for pool name](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9320/commits/88aea5a7a6b4a278e6405b749cd11cdbe25f93e1)

Reply:
And test added in [add test for new pool name sort scope](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9320/commits/39d26681dcb8017eeb0a8c684dd9b3ba50e072bb)